src/main/java/lesson2/problem0/LinkedList.java

```diff
@@ -0,0 +1,364 @@
+package lesson2.problem0;
+
+import javax.naming.OperationNotSupportedException;
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.List;
+import java.util.PriorityQueue;
+import java.util.Set;
+import java.util.Stack;
+
+public class LinkedList<E> {
+    public Link head;
+
+    public LinkedList() {
+        head = null;
+    }
+
+    public Link getHead() {
+        return this.head;
+    }
+
+    public Link setHead(Link newHead) {
+
+        head = newHead;
+
+        return this.head;
+    }
+
+    public boolean isEmpty() {
+        return (head ==null);
+    }
+
+    public void addFirstLink(int value) {
+        Link newLink = new Link(value);
+
+        newLink.next = head;
+
+        head = newLink;
+    }
+
+    public void insert(int value) {
+        Link newLink = new Link(value);
+
+        if (isEmpty()) {
+            this.head = newLink;
+        }
+        else {
+            Link currentNode = head;
+            while (currentNode.next != null) {
+                currentNode = currentNode.next;
+            }
+            currentNode.next = newLink;
+        }
+    }
+
+    public void insertAll(int[] values) {
+        for (int value : values) {
+            Link newLink = new Link(value);
+
+            if (isEmpty()) {
+                this.head = newLink;
+            }
+            else {
+                Link currentNode = head;
+                while (currentNode.next != null) {
+                    currentNode = currentNode.next;
+                }
+                currentNode.next = newLink;
+            }
+        }
+    }
+
+    public Link getLink(int value) {
+        Link linkReference = getHead();
+
+        if (!isEmpty()) {
+            while (linkReference.value != value) {
+                if (linkReference.next == null) {
```

Comment:
You can merge these two ifs into one with an OR operator. Finish loop if either the value is found OR we reached the end of the list.